src/common/state/table.c

```diff
@@ -101,8 +96,11 @@ int64_t table_timeout_handler(event_loop *loop,
     return EVENT_LOOP_TIMER_DONE;
   }
 
-  /* Decrement retry count and try again. */
-  callback_data->retry.num_retries--;
+  /* Decrement retry count and try again. We use -1 to indicate infinite
+   * retries. */
+  if (callback_data->retry.num_retries != -1) {
```

Comment:
Hmm yeah, that makes sense. We should try to test the constant timeout though. Do you have a workload you can try it with?